adafruit_radial_controller/device.py

```diff
@@ -27,55 +25,85 @@ def device(report_id: int) -> usb_hid.Device:
-            0x15, 0x00,       #     Logical Minimum (0)
-            0x25, 0x01,       #     Logical Maximum (1)
-            0x75, 0x01,       #     Report Size (1)
-            0x95, 0x01,       #     Report Count (1)
-            0x81, 0x02,       #     Input (Data,Var,Abs)
-
-            # Padding
-            0x75, 0x07,       #     Report Size (7)
-            0x95, 0x01,       #     Report Count (1)
-            0x81, 0x01,       #     Input (Data,Var,Abs)
-
-            # Rotation
-            0x05, 0x01,       #     Usage Page (Generic Desktop)
-            0x09, 0x37,       #     Usage (Dial)
-            0x55, 0x0f,       #     Unit Exponent (-1)
-            0x65, 0x14,       #     Unit (Degrees, English Rotation)
-            0x36, 0xf0, 0xf1, #     Physical Minimum (-3600)
-            0x46, 0x10, 0x0e, #     Physical Maximum (3600)
-            0x16, 0xf0, 0xf1, #     Logical Minimum (-3600)
-            0x26, 0x10, 0x0e, #     Logical Maximum (3600)
-            0x75, 0x10,       #     Report Size (16)
-            0x95, 0x01,       #     Report Count (1)
-            0x81, 0x06,       #     Input (Data,Var,Rel)
-
-            0xc0,             #   End Collection
-            0xc0,             # End Collection
-
-            # The presence of this (empty) System Control Collection makes the device
-            # appear in /dev/input in Linux, which will help if radial controller
-            # functionality is added in userspace code.
-            0x09, 0x80,       # Usage (System Control)
-            0xa1, 0x01,       # Collection (Application)
-            0xc0,             # End Collection
-            )),
+        report_descriptor=bytes(
+            (
+                0x05,
+                0x01,  # Usage Page (Generic Desktop)
+                0x09,
+                0x0E,  # Usage (System Multi-Axis Controller)
+                0xA1,
+                0x01,  # Collection (Application)
+                0x85,
+                report_id,  #   Report Id (Radial Controller)
+                0x05,
+                0x0D,  #   Usage Page (Digitizers)
+                0x09,
+                0x21,  #   Usage (Puck)
+                0xA1,
+                0x00,  #   Collection (Physical)
+                # The Microsoft example packs the report into 1 bit and 15 bits,
+                # but that's more work. We'll just use 1 bit of one byte and then 16 bits.
+                # Button
+                0x05,
+                0x09,  #     Usage Page (Buttons)
+                0x09,
+                0x01,  #     Usage (Button 1)
+                0x15,
+                0x00,  #     Logical Minimum (0)
+                0x25,
+                0x01,  #     Logical Maximum (1)
+                0x75,
+                0x01,  #     Report Size (1)
+                0x95,
+                0x01,  #     Report Count (1)
+                0x81,
+                0x02,  #     Input (Data,Var,Abs)
+                # Padding
+                0x75,
```

Comment:
I think this was a better format before. I'm going to circle back and undo the format change to this chunk of values and figure out how to get ruff formatter to ignore this section.